### PR TITLE
Override EXPECT_NEW_PACKAGES if we revoked key packages (EXPOSUREAPP-3455)

### PR DESCRIPTION
**Pull request by d4rken**
We need to set `force index lookup` to `true` if we revoked any local key packages, as we might still have the newest package, so the check (i.e. `expectNewDayPackages`) could return `false` because we revoked a key package that wasn't the newest one (which is what the check mostly looks at).

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/BaseKeyPackageSyncTool.kt [hunk 1]
```diff
@@ -14,18 +14,28 @@ open class BaseKeyPackageSyncTool(
     private val tag: String
 ) {
 
-    internal suspend fun revokeCachedKeys(revokedKeyPackages: Collection<KeyDownloadConfig.RevokedKeyPackage>) {
+    /**
+     * Returns true if any of our cached keys were revoked
+     */
+    internal suspend fun revokeCachedKeys(
```
Comment (ralfgehrer) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/BaseKeyPackageSyncTool.kt:20]: Optional: add some tests for this

**Review by ralfgehrer — APPROVED** [review]
Looks good.

**Review by kolyaopahle — APPROVED** [review]
LGTM 👍

**Review by harambasicluka — COMMENTED** [review]
Tested on device

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/BaseKeyPackageSyncTool.kt (hunk 1 above)
Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/BaseKeyPackageSyncTool.kt:20]: https://github.com/corona-warn-app/cwa-app-android/pull/1580

**Pull request merged**